Castling

rjdellecese opened the pull request:
@lasley Pretty straightforward, I think. I split up the methods according to either kingside and queenside castling, and didn't actually have to worry about the differences between white/black as everything else was calculated relative to the king's y-position. Checking that both the king and the rook the king was castling with haven't moved yet ensures that there is no possibility that the king can castle on ranks other than the one it begins on (that is, I don't need to check that the y-position is 1 if white or 8 if black).

lasley commented on coverage/.last_run.json:
```diff
@@ -0,0 +1,5 @@
+{
```
Remove entire coverage directory, and add `coverage` to gitignore to avoid this

lasley commented:
Straightforward indeed. Excellent tests! 

One comment about the coverage dir then we're :+1: IMO

rjdellecese commented:
@lasley Woohoo, thanks! I deleted the `coverage` dir, and looks like it was already in gitignore from the merge with master. I also noticed an empty game factory I never ended up using, so I deleted that as well.
